src/AdbackAnalytics/Generator/AutopromoBannerScriptGenerator.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Dekalee\AdbackAnalytics\Generator;
+
+/**
+ * Class AutopromoBannerScriptGenerator
+ */
+class AutopromoBannerScriptGenerator extends AbstractScriptGenerator implements ScriptGeneratorInterface
+{
+    /**
+     * @param int $id
+     *
+     * @return string
+     */
+    public function generate($id)
```

Comment:
in this method I would set `$id = null` and throw an explicit exception if the id is null